app/components/edit/EditAddress.jsx

```diff
@@ -1,129 +1,252 @@
+            <span className={s.formControlGroupLabel}>Street Address</span>
+            <input
+              className={s.inputTextbox}
+              type="text"
+              name="address_1"
+              placeholder="Address"
+              required
+              defaultValue={defaultData.address_1}
+            />
+            <input
+              className={s.inputTextbox}
+              type="text"
+              name="address_2"
+              placeholder="Apartment, suite, unit, building, floor, etc."
+              defaultValue={defaultData.address_2}
+            />
+          </div>
+          <label className={s.inputLabel}>
+            City
+            <input
+              className={s.inputTextbox}
+              type="text"
+              name="city"
+              placeholder="City"
+              required
+              defaultValue={defaultData.city}
+            />
+          </label>
+          <label className={s.inputLabel}>
+            State
+            <input
+              className={s.inputTextbox}
+              type="text"
+              name="state_province"
+              placeholder="State"
+              required
+              defaultValue={defaultData.state_province}
+            />
+          </label>
+          <label className={s.inputLabel}>
+            Zip code
+            <input
+              className={s.inputTextbox}
+              type="text"
+              name="postal_code"
+              placeholder="Zip Code"
+              required
+              defaultValue={defaultData.postal_code}
+            />
+          </label>
+        </div>
+        <div className={s.actionRow}>
+          <button
+            className={s.cancelButton}
+            type="button"
+            onClick={onRequestClose}
+          >
+            ← Cancel
+          </button>
+          <input
+            className={s.addButton}
+            type="submit"
+            value={submitButtonText}
+          />
+        </div>
+      </form>
+    </ReactModal>
   );
 };
 
-EditAddress.propTypes = {
-  item: PropTypes.object.isRequired,
-  handleChange: PropTypes.func.isRequired,
+EditAddressModal.propTypes = {
+  isOpen: PropTypes.bool.isRequired,
 };
 
+/** Format address as a single line. */
+const compactAddressDisplay = ({
+  address_1, address_2, address_3, address_4, city, state_province, postal_code,
+}) => {
+  // No comma between state and postal code
+  const state_postal = `${state_province} ${postal_code}`;
+  const lines = [address_1, address_2, address_3, address_4, city, state_postal];
+  return lines.filter(x => x).join(', ');
+};
+
+const AddressListItem = ({
+  displayIndex, address, onEdit, onRemove,
+}) => (
+  <div className={s.listItemContainer}>
+    <div className={s.listItemIndex}>
+      {`${displayIndex}.`}
+    </div>
+    <div className={s.listItemName}>{address.name}</div>
+    <div className={s.listItemAddress}>{compactAddressDisplay(address)}</div>
```

Comment:
I'm not sure we actually want to do that, since we actually want to remove all line breaks here to make it all fit in one line. I'm using `white-space: nowrap`, not `white-space: pre`, for the CSS of `listItemAddress`.

I'm going to leave this alone, but let me know if you feel strongly otherwise.